kie-server-parent/kie-server-services/kie-server-services-jbpm/src/main/java/org/kie/server/services/jbpm/security/ElytronUserGroupAdapter.java

```diff
@@ -23,44 +23,63 @@
 import java.util.stream.StreamSupport;
 
 import org.jbpm.services.task.identity.adapter.UserGroupAdapter;
+import org.kie.server.services.impl.security.ElytronIdentityProvider;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.wildfly.security.auth.server.RealmIdentity;
 import org.wildfly.security.auth.server.RealmUnavailableException;
 import org.wildfly.security.auth.server.SecurityDomain;
 import org.wildfly.security.auth.server.SecurityIdentity;
-import org.wildfly.security.authz.AuthorizationFailureException;
 
 public class ElytronUserGroupAdapter implements UserGroupAdapter {
 
     private static final Logger logger = LoggerFactory.getLogger(ElytronUserGroupAdapter.class);
+    private Class<?> authorizationFailureExceptionClass = null;
+
+    public ElytronUserGroupAdapter() {
+        try {
+            this.authorizationFailureExceptionClass = Class.forName("org.wildfly.security.authz.AuthorizationFailureException");
+        } catch (Exception var2) {
+            logger.info("Unable to find org.wildfly.security.authz.AuthorizationFailureException, disabling elytron adapter");
+        }
+    }
+
+    protected boolean isActive() {
+        return this.authorizationFailureExceptionClass != null;
+    }
 
     @Override
     public List<String> getGroupsForUser(String userId) {
```

Comment:
`Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed.` https://sonarcloud.io/project/issues?resolved=false&severities=CRITICAL&pullRequest=2967&id=org.drools%3Adroolsjbpm-integration&open=AYhzEjM4inEykiRLLCBi